bundles/ch.elexis.core.ui/src/ch/elexis/core/ui/preferences/UserCasePreferences.java

```diff
@@ -358,25 +399,51 @@ public void widgetDefaultSelected(SelectionEvent e) {
 		return suParent;
 	}
 
+	private void setupTableWithColors() {
+		List<IBillingSystem> allSystems = BillingSystemServiceHolder.get().getBillingSystems();
+		List<String> systemNames = allSystems.stream().map(IBillingSystem::getName).toList();
+		String[] entries = sortBillingSystems(systemNames.toArray(new String[0]), topItemsLinkedList, true);
+		sorterList2Viewer.setInput(entries);
+	}
```

Comment:
Bitte in den Viewer direkt die `IBillingSystem` Objekte als Input verwenden. Für den Separator sollte es möglich sein ein `SeparatorIBillingSystem implements IBillingSystem` Klasse zu verwenden.

Für die Sortierung eienen `Comparator` mit `sorterList2Viewer.setComparator` hinzufügen. Hier muss dann auch die Logik für pre sorted rein.

Beim `LabelProvider` kommen dann die `IBillingSystem` Objekte als element rein.

Zum Selektieren kann dann `sorterList2Viewer.setSelection` mit einer `StructuredSelection` mit dem `IBillingSystem` Objekt verwendet werden. Dadurch sollte die `indexOfTableItem` Methode weg fallen.